tests/functional/disaster-recovery/metro-dr/test_app_failover_and_relocate_when_entire_one_zone_down.py

```diff
@@ -0,0 +1,378 @@
+        logger.info(
+            "All nodes from active hub zone are powered off, "
+            f"wait {wait_time} seconds before restoring in passive hub"
+        )
+
+        # Restore new hub
+        restore_backup()
+        wait_time = 300
+        logger.info(f"Wait {wait_time} until restores are taken ")
+        time.sleep(wait_time)
+
+        # Verify the restore is completed
+        verify_restore_is_completed()
+
+        # Validate the clusters are imported
+        clusters = [self.primary_cluster_name, secondary_cluster_name]
+        for cluster in clusters:
+            for sample in TimeoutSampler(
+                timeout=1800,
+                sleep=60,
+                func=validate_cluster_import,
+                cluster_name=cluster,
+                switch_ctx=get_passive_acm_index(),
+            ):
+                if sample:
+                    logger.info(
+                        f"Cluster: {cluster} successfully imported post hub recovery"
+                    )
+                    # Validate klusterlet addons are running on managed cluster
+                    config.switch_to_cluster_by_name(cluster)
+                    wait_for_pods_to_be_running(
+                        namespace=constants.ACM_ADDONS_NAMESPACE, timeout=300, sleep=15
+                    )
+                    break
+                else:
+                    logger.error(
+                        f"import of cluster: {cluster} failed post hub recovery"
+                    )
+                    raise UnexpectedBehaviour(
+                        f"import of cluster: {cluster} failed post hub recovery"
+                    )
+
+        # Wait or verify the drpolicy is in validated state
+        verify_drpolicy_cli(switch_ctx=get_passive_acm_index())
+
+        # ToDo: Deploy application in both managed cluster and
+        #  to verify the applications are present in secondary cluster
+
+        # Fenced the primary managed cluster
+        enable_fence(
+            drcluster_name=self.primary_cluster_name,
+            switch_ctx=get_passive_acm_index(),
+        )
+        # Verify the primary managed cluster is in Fenced state
+        verify_fence_state(
+            drcluster_name=self.primary_cluster_name,
+            state=constants.ACTION_FENCE,
+            switch_ctx=get_passive_acm_index(),
+        )
+
+        # Application Failover to Secondary managed cluster
+        if (
+            config.RUN.get("mdr_failover_via_ui")
+            and self.workload_type == constants.SUBSCRIPTION
+        ):
+            logger.info("Start the process of Failover from ACM UI")
+            config.switch_ctx(get_passive_acm_index())
+            failover_relocate_ui(
+                acm_obj,
+                workload_to_move=f"{workloads[0].workload_name}-1",
+                policy_name=workloads[0].dr_policy_name,
+                failover_or_preferred_cluster=secondary_cluster_name,
+            )
```

Comment:
Need to add wa for sub app pods to show up after failover in ACM 2.11:
`Workaround:
Add a new label with any value to the AppSub on the hub.
command:
oc edit appsub -n <appsub-namespace> <appsub>-subscription-1
  labels:
    app: vm-pvc-acm-sub1
    app.kubernetes.io/part-of: vm-pvc-acm-sub1
    apps.open-cluster-management.io/reconcile-rate: medium
    test: test1`

Reply:
Added now